test/k8s/simple_test.go

```diff
@@ -20,137 +21,147 @@ import (
 
-	BeforeAll(func() {
-		opts := []KumaDeploymentOption{
-			WithCtlOpts(map[string]string{
-				"--set": "" +
-					"kuma.controlPlane.resources.requests.cpu=1," +
-					"kuma.controlPlane.resources.requests.memory=2Gi," +
-					"kuma.controlPlane.resources.limits.memory=32Gi",
-				"--env-var": "" +
-					"KUMA_RUNTIME_KUBERNETES_LEADER_ELECTION_LEASE_DURATION=100s," +
-					"KUMA_RUNTIME_KUBERNETES_LEADER_ELECTION_RENEW_DEADLINE=80s",
-			}),
-		}
+		var svcGraph graph.Services
 
-		alternativeContainerRegistry, _ = os.LookupEnv("ALTERNATIVE_CONTAINER_REGISTRY")
+		var alternativeContainerRegistry string
 
-		if alternativeContainerRegistry != "" {
-			opts = append(opts,
+		BeforeAll(func() {
+			envVars := append([]string{
+				"KUMA_RUNTIME_KUBERNETES_LEADER_ELECTION_LEASE_DURATION=100s",
+				"KUMA_RUNTIME_KUBERNETES_LEADER_ELECTION_RENEW_DEADLINE=80s",
+			}, fl.envVars...)
+			helmFlags := append([]string{
+				"kuma.controlPlane.resources.requests.cpu=1",
+				"kuma.controlPlane.resources.requests.memory=2Gi",
+				"kuma.controlPlane.resources.limits.memory=32Gi",
+			}, fl.helmFlags...)
+
+			opts := []KumaDeploymentOption{
 				WithCtlOpts(map[string]string{
-					"--dataplane-registry": alternativeContainerRegistry,
-				}))
-		}
-
-		opts = append(opts,
-			WithCtlOpts(map[string]string{
-				"--license-path": kmeshLicense,
-			}))
-
-		err := NewClusterSetup().
-			Install(Kuma(core.Zone, opts...)).
-			Install(NamespaceWithSidecarInjection(TestNamespace)).
-			Setup(cluster)
-		Expect(err).ToNot(HaveOccurred())
-
-		num := requireVar("PERF_TEST_NUM_SERVICES")
-		i, err := strconv.Atoi(num)
-		Expect(err).ToNot(HaveOccurred(), "invalid value of PERF_TEST_NUM_SERVICES")
-		numServices = i
-
-		num = requireVar("PERF_TEST_INSTANCES_PER_SERVICE")
-		i, err = strconv.Atoi(num)
-		Expect(err).ToNot(HaveOccurred(), "invalid value of PERF_TEST_INSTANCES_PER_SERVICE")
-		instancesPerService = i
-
-		svcGraph = graph.GenerateRandomServiceMesh(872835240, numServices, 50, instancesPerService, instancesPerService)
-	})
-
-	BeforeEach(func() {
-		Expect(framework.ReportSpecStart(cluster)).To(Succeed())
-		start = time.Now()
-	})
-
-	AfterEach(func() {
-		endpoint := cluster.GetPortForward("prometheus-server").ApiServerEndpoint
-		promClient, err := framework.NewPromClient(fmt.Sprintf("http://%s", endpoint))
-		Expect(err).ToNot(HaveOccurred())
-
-		stopCh := make(chan struct{})
-		metricCh := make(chan int)
-		errCh := make(chan error)
-
-		go framework.WatchXdsDeliveryCount(promClient, stopCh, metricCh, errCh)
-		defer close(stopCh)
-
-	Loop:
-		for {
-			select {
-			case <-metricCh:
-			case err := <-errCh:
-				Fail(err.Error())
-			case <-time.After(stabilizationSleep):
-				break Loop
+					"--set":          strings.Join(helmFlags, ","),
+					"--env-var":      strings.Join(envVars, ","),
+					"--license-path": kmeshLicense,
+				}),
+			}
+
+			alternativeContainerRegistry, _ = os.LookupEnv("ALTERNATIVE_CONTAINER_REGISTRY")
```

Comment:
May not be relating to this PR, but should we remove this `alternativeContainerRegistry` now? Looks like it's pointing to a personal image repository defaultly. I guess it's an ex-member of the team?